### PR TITLE
feat(AccessCode): add ability to delete

### LINKED ISSUE #327: Access Code - ability to delete codes
Extension of https://github.com/seamapi/react/issues/236, we should be able to delete codes.

### PR DESCRIPTION
**Pull request by mikewuu**
Closes #327 

### Dependencies

- https://github.com/seamapi/fake-seam-connect/pull/48

### Notable things

- Added a `disableDeleteAccessCode` prop, similar to `disableEditAccessCode`.
- Added **View code details menu item**

### Screenshots

CleanShot 2023-08-01 at 01 31 12@2x

https://github.com/seamapi/react/assets/11449462/7dc34ac8-99ab-4b94-b696-ef9578d7eace

### CODE REVIEW

File: src/lib/seam/components/AccessCodeTable/AccessCodeMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  onEdit,
+}: AccessCodeMenuProps): JSX.Element {
+  const [deleteConfirmationVisible, toggleDeleteConfirmation] = useToggle()
+
+  const deleteAccessCode = useDeleteAccessCode()
+
+  if (deleteConfirmationVisible) {
+    return (
+      <div className='seam-delete-confirmation'>
+        <span>Delete this code and data?</span>
```
Comment (razor-x) [inline src/lib/seam/components/AccessCodeTable/AccessCodeMenu.tsx:46]: Move to `t.`

**Review by razor-x — APPROVED** [review]
Nice! Just a few minor suggestions.

File: src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx [hunk 2]
```diff
@@ -17,6 +18,7 @@ import { ContentHeader } from 'lib/ui/layout/ContentHeader.js'
 import { useIsDateInPast } from 'lib/ui/use-is-date-in-past.js'
 
 const disableEditAccessCode = true
+const disableDeleteAccessCode = true
```
[the remark this answers is not in the document]
Comment (razor-x) [reply src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx:21]: Can you PR to update the docs on these pages after this is merged? https://docs.seam.co/latest/seam-components/react-components

File: src/lib/ui/Menu/MenuItem.tsx [hunk 3]
```diff
@@ -1,20 +1,31 @@
+import classNames from 'classnames'
 import type { PropsWithChildren } from 'react'
 
 import { useMenu } from 'lib/ui/Menu/Menu.js'
 
 interface MenuItemProps extends PropsWithChildren {
-  onClick: () => void
+  onClick: (event: React.MouseEvent) => void
+  className?: string
+  disableCloseOnClick?: boolean
```
[the remark this answers is not in the document]
Comment (mikewuu) [reply src/lib/ui/Menu/MenuItem.tsx]: Renamed in https://github.com/seamapi/react/pull/328/commits/02434aca56bf2a54359b912a4b43e33114574768

File: src/lib/ui/Menu/MenuItem.tsx [hunk 4]
```diff
@@ -1,20 +1,31 @@
...
 }
 
-export function MenuItem({ onClick, children }: MenuItemProps): JSX.Element {
+export function MenuItem({
+  onClick,
+  children,
+  className,
+  disableCloseOnClick = false,
```
[the remark this answers is not in the document]
Comment (mikewuu) [reply src/lib/ui/Menu/MenuItem.tsx]: renamed in https://github.com/seamapi/react/pull/328/commits/02434aca56bf2a54359b912a4b43e33114574768

File: src/lib/ui/Menu/MenuItem.tsx [hunk 5]
```diff
@@ -1,20 +1,31 @@
...
     <div
-      className='seam-menu-item'
-      onClick={() => {
-        onClick()
-        closeMenu()
+      className={classNames('seam-menu-item', className)}
+      onClick={(event: React.MouseEvent) => {
+        onClick(event)
+
+        if (!disableCloseOnClick) {
```
[the remark this answers is not in the document]
Comment (mikewuu) [reply src/lib/ui/Menu/MenuItem.tsx]: renamed in https://github.com/seamapi/react/pull/328/commits/02434aca56bf2a54359b912a4b43e33114574768

File: src/lib/seam/components/AccessCodeTable/AccessCodeMenu.tsx (hunk 1 above)
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeTable/AccessCodeMenu.tsx]: updated in https://github.com/seamapi/react/pull/328/commits/3b08def1bc6db9e637ac008a38c0db44e07fb5aa 😃

File: src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx]: Updated in https://github.com/seamapi/react/pull/328/commits/e2b643445d06d8afba727e45b68f38f37e6d73df

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx [hunk 6]
```diff
@@ -16,27 +12,22 @@ import {
...
 import { TableTitle } from 'lib/ui/Table/TableTitle.js'
 import { SearchTextField } from 'lib/ui/TextField/SearchTextField.js'
 import { Caption } from 'lib/ui/typography/Caption.js'
-import { Title } from 'lib/ui/typography/Title.js'
 import { useToggle } from 'lib/ui/use-toggle.js'
 
 const disableCreateAccessCode = true
-const disableEditAccessCode = false
+const disableEditAccessCode = true
+const disableDeleteAccessCode = true
```
[the remark this answers is not in the document]
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx]: Updated in https://github.com/seamapi/react/pull/328/commits/e2b643445d06d8afba727e45b68f38f37e6d73df

**Comment by dawnho** [comment]
@miken-seam, when we delete an access code, its status switched over to "removing". I wonder how we can show this visually on a table?

**Review by dawnho — APPROVED** [review]
Changes look good. I had a comment about showing the code removal state, but we can handle that in a separate issue

**Pull request merged**